CordovaIsAvailableJob fails on Windows 8, Eclipse Neon

When trying to create a new Hybrid Mobile project, the whizard fails to recognice cordovals version:
```

> 
> java.lang.IllegalArgumentException: invalid version "Microsoft Windows [Version 6.3.9600]
> (c) 2013 Microsoft Corporation. Alle Rechte vorbehalten.
> 
> D:\Programme\eclipse_Neon>cordova -version
> 7.0.1
> 
> D:\Programme\eclipse_Neon>exit": non-numeric "Microsoft Windows [Version 6"
> 	at org.osgi.framework.Version.parseInt(Version.java:170)
> 	at org.osgi.framework.Version.<init>(Version.java:126)
> 	at org.osgi.framework.Version.valueOf(Version.java:257)
> 	at org.osgi.framework.Version.parseVersion(Version.java:228)
> 	at org.eclipse.thym.ui.wizard.project.WizardNewHybridProjectCreationPage$CordovaIsAvailableJob.run(WizardNewHybridProjectCreationPage.java:232)
> 	at org.eclipse.core.internal.jobs.Worker.run(Worker.java:55)
> Caused by: java.lang.NumberFormatException: For input string: "Microsoft Windows [Version 6"
> 	at java.lang.NumberFormatException.forInputString(Unknown Source)
> 	at java.lang.Integer.parseInt(Unknown Source)
> 	at java.lang.Integer.parseInt(Unknown Source)
> 	at org.osgi.framework.Version.parseInt(Version.java:168)
> 	... 5 more
> 


Environment:
I'm using latest developer snapshot of thym as this is able to download stuff from behind a proxy.

> 
> eclipse.buildId=4.6.1.M20160907-1200
> java.version=1.8.0_101
> java.vendor=Oracle Corporation
> BootLoader constants: OS=win32, ARCH=x86_64, WS=win32, NL=de_DE
> Framework arguments:  -product org.eclipse.epp.package.jee.product -product org.eclipse.epp.package.jee.product
> Command-line arguments:  -os win32 -ws win32 -arch x86_64 -product org.eclipse.epp.package.jee.product -data file:/D:/Development/Eclipse/Neon/Mobile/ -product org.eclipse.epp.package.jee.product
>

--- a/plugins/org.eclipse.thym.core/src/org/eclipse/thym/core/internal/cordova/ErrorDetectingCLIResult.java
+++ b/plugins/org.eclipse.thym.core/src/org/eclipse/thym/core/internal/cordova/ErrorDetectingCLIResult.java
@@ -1,5 +1,5 @@
 /*******************************************************************************
- * Copyright (c) 2015, 2016 Red Hat, Inc. 
+ * Copyright (c) 2015, 2017 Red Hat, Inc. 
  * All rights reserved. This program and the accompanying materials
  * are made available under the terms of the Eclipse Public License v1.0
  * which accompanies this distribution, and is available at
@@ -20,6 +20,7 @@ public class ErrorDetectingCLIResult extends CordovaCLIResult{
 	private static final String ERROR_PREFIX = "Error:";
 	private StringBuffer errorMessage = new StringBuffer();
 	private int errorCode = CordovaCLIErrors.ERROR_GENERAL; 
+	public static final String CORDOVA_NOT_FOUND="Cordova not found, please run 'npm install -g cordova' on a command line";
 	
 	public ErrorDetectingCLIResult(String message) {
 		super(message);
@@ -44,7 +45,7 @@ public class ErrorDetectingCLIResult extends CordovaCLIResult{
 				errorMessage = errorMessage.append(line.substring(ERROR_PREFIX.length(), line.length()).trim());
 			}else if(line.contains("command not found") || line.contains("is not recognized as an internal or external command")){
 				error = true;
-				errorMessage.append("Cordova not found, please run 'npm install -g cordova' on a command line to install Cordova globally");
+				errorMessage.append(CORDOVA_NOT_FOUND);
 				errorCode = CordovaCLIErrors.ERROR_COMMAND_MISSING;
 			}
 			else{

--- a/plugins/org.eclipse.thym.ui/src/org/eclipse/thym/ui/wizard/project/WizardNewHybridProjectCreationPage.java
+++ b/plugins/org.eclipse.thym.ui/src/org/eclipse/thym/ui/wizard/project/WizardNewHybridProjectCreationPage.java
@@ -13,7 +13,6 @@ package org.eclipse.thym.ui.wizard.project;
 import static org.eclipse.jface.dialogs.Dialog.DLG_IMG_MESSAGE_WARNING;
 import static org.eclipse.jface.dialogs.Dialog.DLG_IMG_MESSAGE_ERROR;
 
-import org.eclipse.core.runtime.CoreException;
 import org.eclipse.core.runtime.IProgressMonitor;
 import org.eclipse.core.runtime.IStatus;
 import org.eclipse.core.runtime.Status;
@@ -27,6 +26,8 @@ import org.eclipse.swt.SWT;
 import org.eclipse.swt.custom.CLabel;
 import org.eclipse.swt.events.ModifyEvent;
 import org.eclipse.swt.events.ModifyListener;
+import org.eclipse.swt.graphics.Color;
+import org.eclipse.swt.graphics.Image;
 import org.eclipse.swt.layout.GridData;
 import org.eclipse.swt.layout.GridLayout;
 import org.eclipse.swt.widgets.Composite;
@@ -40,6 +41,7 @@ import org.eclipse.ui.dialogs.WorkingSetGroup;
 import org.osgi.framework.Version;
 import org.eclipse.thym.core.HybridProjectConventions;
 import org.eclipse.thym.core.internal.cordova.CordovaCLI;
+import org.eclipse.thym.core.internal.cordova.CordovaCLIErrors;
 import org.eclipse.thym.core.internal.cordova.ErrorDetectingCLIResult;
 import org.eclipse.thym.ui.HybridUI;
 
@@ -139,7 +141,7 @@ public class WizardNewHybridProjectCreationPage extends WizardNewProjectCreation
 			return false;
 		}
 		if(!cordovaFound){
-			setMessage("Cordova not found, please run 'npm install -g cordova' on a command line", ERROR);
+			setMessage(ErrorDetectingCLIResult.CORDOVA_NOT_FOUND, ERROR);
 			this.getContainer().updateMessage();
 			return false;
 		}
@@ -228,13 +230,17 @@ public class WizardNewHybridProjectCreationPage extends WizardNewProjectCreation
 		protected IStatus run(IProgressMonitor monitor) {
 			try {
 				ErrorDetectingCLIResult result = new CordovaCLI().version(monitor).convertTo(ErrorDetectingCLIResult.class);
+				if(result.asStatus().getCode() == CordovaCLIErrors.ERROR_COMMAND_MISSING){
+					return Status.OK_STATUS;
+				}
 				if(result.asStatus().isOK()){
 					cordovaVersion = Version.parseVersion(result.getMessage()).toString();
+					return Status.OK_STATUS;
 				}
-			} catch (CoreException e) {
+			} catch (Exception e) {
 				HybridUI.log(WARNING, "Unable to determine if cordova is available", e);
 			}
-			return Status.OK_STATUS;
+			return new Status(Status.WARNING, HybridUI.PLUGIN_ID, "");
 			
 		}
 		
@@ -254,19 +260,29 @@ public class WizardNewHybridProjectCreationPage extends WizardNewProjectCreation
 			public void done(IJobChangeEvent event) {
 				if(!getControl().isDisposed()){
 					final Display display = getControl().getDisplay();
+					final int severity = event.getJob().getResult().getSeverity();
 					display.syncExec(new Runnable() {
 						@Override
 						public void run() {
-							if(cordovaIsAvailableJob.getCordovaVersion() != null){
-								cordovaLabel.setText(cordovaIsAvailableJob.getCordovaVersion());
-								cordovaLabel.setImage(null);
-								cordovaFound = true;
+							if(severity == Status.OK){
+								if(cordovaIsAvailableJob.getCordovaVersion() != null){
+									updateCordovaLabel(cordovaIsAvailableJob.getCordovaVersion(), null, null, true);
+								} else {
+									updateCordovaLabel(ErrorDetectingCLIResult.CORDOVA_NOT_FOUND, 
+											JFaceResources.getImage(DLG_IMG_MESSAGE_ERROR), 
+											getControl().getDisplay().getSystemColor(SWT.COLOR_RED), false);
+								}
 							} else {
-								cordovaLabel.setText("Cordova not found, please run 'npm install -g cordova' on a command line");
-								cordovaLabel.setImage(JFaceResources.getImage(DLG_IMG_MESSAGE_ERROR));
-								cordovaLabel.setForeground(getControl().getDisplay().getSystemColor(SWT.COLOR_RED));
-								cordovaFound = false;
+								updateCordovaLabel("Unable to determine if cordova is available", 
+										JFaceResources.getImage(DLG_IMG_MESSAGE_WARNING), null, true);
 							}
+						}
+						
+						private void updateCordovaLabel(String text, Image image, Color foreground , boolean found){
+							cordovaLabel.setText(text);
+							cordovaLabel.setImage(image);
+							cordovaLabel.setForeground(foreground);
+							cordovaFound = found;
 							setPageComplete(validatePage());
 						}
 					});

